lib/Parse/ParseDecl.cpp

```diff
@@ -1092,7 +1119,23 @@ static bool parseQualifiedDeclName(Parser &P, Diag<> nameParseError,
       Parser::DeclNameFlag::AllowOperators);
   // The base type is optional, but the final unqualified declaration name is
   // not. If name could not be parsed, return true for error.
-  return !original.Name;
+  if (!original.Name)
+    return true;
+  }
+
+  // Parse to see if this is an accessor and set it's type.  This is an optional field.
```

Comment:
If this isn't tested already, could you please add a `@derivative` attribute test case referencing the accessor of a non-storage declaration? This should be diagnosed.
```swift
func function(_ x: Float) -> Float {
  x
}

// Test invalid reference to an accessor of a non-storage declaration.
@derivative(of: function(_:).get)
func vjpFunction(_ x: Float) -> (value: Float, pullback: (Float) -> Float) {
  fatalError()
}
```

Reply:
Added.  Good one.